core/code/portal_detail_display.js

```diff
@@ -78,27 +78,30 @@ window.renderPortalDetails = function(guid) {
   levelDetails = "Level " + levelDetails;
 
 
-  var linkDetails = [];
+  var linkDetails = $('<div>', { class: 'linkdetails' });
 
-  var posOnClick = 'window.showPortalPosLinks('+lat+','+lng+',\''+escapeJavascriptString(title)+'\')';
-  var permalinkUrl = window.makePermalink([lat,lng]);
+  var posOnClick = window.showPortalPosLinks.bind(this,lat,lng,title);
 
   if (typeof android !== 'undefined' && android && android.intentPosLink) {
     // android devices. one share link option - and the android app provides an interface to share the URL,
     // share as a geo: intent (navigation via google maps), etc
 
-    var shareLink = $('<div>').html( $('<a>').attr({onclick:posOnClick}).text('Share portal') ).html();
-    linkDetails.push('<aside>'+shareLink+'</aside>');
+    var shareLink = $('<a>').text('Share portal').click(posOnClick);
+    linkDetails.append($('<aside>').append($('<div>').append(shareLink)));
 
   } else {
     // non-android - a permalink for the portal
-    var permaHtml = $('<div>').html( $('<a>').attr({href:permalinkUrl, title:'Create a URL link to this portal'}).text('Portal link') ).html();
-    linkDetails.push ( '<aside>'+permaHtml+'</aside>' );
+    var permaHtml = $('<a>').attr({
```

Comment:
Note: you can even skip the "attr" here, like:
`$("<a>", {href: ...; title: "..."; text: "ffff"; click: ...})`

Reply:
> Note: you can even skip the "attr" here, like: [...]

Yeah, but (imho) in that case readability would be worse.

Reply:
Yeah, may be. Matter of taste.